components/raftstore/src/store/fsm/store.rs

```diff
@@ -2509,6 +2534,53 @@ impl<'a, EK: KvEngine, ER: RaftEngine, T: Transport> StoreFsmDelegate<'a, EK, ER
         }
     }
 
+    /// Use ranges assigned to each region as increments for full compaction.
+    fn ranges_for_full_compact(&self) -> Vec<(Vec<u8>, Vec<u8>)> {
+        let meta = self.ctx.store_meta.lock().unwrap();
+        let mut ranges = Vec::with_capacity(meta.region_ranges.len());
+
+        for region_id in meta.region_ranges.values() {
```

Comment:
Though some existing code has used this pattern before, I think it would be better to just iterate the meta.regions.value to get the region object directly. There's no need to do a lookup by each region id.

Reply:
done.